torch/nn/modules/batchnorm.py

```diff
@@ -38,27 +38,37 @@ def __init__(self, num_features, eps=1e-5, momentum=0.1, affine=True,
             self.register_parameter('running_mean', None)
             self.register_parameter('running_var', None)
             self.register_parameter('num_batches_tracked', None)
+        self.num_batches_tracked_scalar = -1
         self.reset_parameters()
 
     def reset_running_stats(self):
         if self.track_running_stats:
             self.running_mean.zero_()
             self.running_var.fill_(1)
             self.num_batches_tracked.zero_()
+            self.num_batches_tracked_scalar = 0
 
     def reset_parameters(self):
         self.reset_running_stats()
+        if self.track_running_stats:
+            self.num_batches_tracked_scalar = int(self.num_batches_tracked)
         if self.affine:
             init.ones_(self.weight)
             init.zeros_(self.bias)
 
+    @property
+    def num_batches_tracked(self):
```

Comment:
can you have both a buffer and a @property with the same name? That's pretty confusing